components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -797,11 +804,18 @@ public static API prepareToCreateAPIByDTO(APIDTO body, APIProvider apiProvider,
         } else if (body.getContext().endsWith("/")) {
             throw new APIManagementException("Context cannot end with '/' character", ExceptionCodes.INVALID_CONTEXT);
         }
-        if (apiProvider.isApiNameWithDifferentCaseExist(body.getName())) {
-            throw new APIManagementException(
-                    "Error occurred while adding API. API with name " + body.getName() + " already exists.",
-                    ExceptionCodes.API_ALREADY_EXISTS);
+
+
+        if (organizationId != null) {
+
+        } else {
+            if (apiProvider.isApiNameWithDifferentCaseExist(body.getName())) {
```

Comment:
can we combine these into one If?

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9609/commits/adb1f3dbfaedb3a5a796cfdaab55385b2e4dfb1b